models/setup_for_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/go-xorm/core"
+	"github.com/go-xorm/xorm"
+	_ "github.com/mattn/go-sqlite3" // for the test engine
+)
+
+func TestMain(m *testing.M) {
+	if err := CreateTestEngine(); err != nil {
+		os.Exit(1)
+	}
+	fmt.Print("Inside TestMain!\n")
+	os.Exit(m.Run())
+}
+
+// CreateTestEngine create an xorm engine for testing
+func CreateTestEngine() error {
```

Comment:
Not a huge fan of using custom db-setups for this, though seeing as `models.LoadConfigs` requires the use of `app.ini` it can't easily be done w/o that...